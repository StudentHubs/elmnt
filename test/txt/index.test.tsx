import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Txt from '../../src/txt';

describe('txt', () => {
  it('empty', () => {
    expect(renderer.create(<Txt />).toJSON()).toMatchSnapshot();
  });

  // it('content only', () => {
  //   expect(renderer.create(
  //     <Txt>Hello World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('placeholder only', () => {
  //   expect(renderer.create(
  //     <Txt placeholder="Hello World!" />
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('content with style', () => {
  //   expect(renderer.create(
  //     <Txt style={{ fontSize: 16, color: 'red', width: 100 }}>Hello World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('placeholder with style', () => {
  //   expect(renderer.create(
  //     <Txt placeholder="Hello World!" style={{ color: 'red', placeholder: { color: 'blue' } }} />
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('content with placeholder style', () => {
  //   expect(renderer.create(
  //     <Txt style={{ color: 'red', placeholder: { color: 'blue' } }}>Hello World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange only', () => {
  //   expect(renderer.create(
  //     <Txt onTextChange={console.log} />
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with content', () => {
  //   expect(renderer.create(
  //     <Txt onTextChange={console.log}>Hello World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with placeholder', () => {
  //   expect(renderer.create(
  //     <Txt placeholder="Hello World!" onTextChange={console.log} />
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with placeholder and content', () => {
  //   expect(renderer.create(
  //     <Txt placeholder="Hello!" onTextChange={console.log}>World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with rows', () => {
  //   expect(renderer.create(
  //     <Txt rows={3} onTextChange={console.log} />
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with rows and content', () => {
  //   expect(renderer.create(
  //     <Txt rows={3} onTextChange={console.log}>Hello World!</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });

  // it('onTextChange with rows and multi-line content', () => {
  //   expect(renderer.create(
  //     <Txt rows={3} onTextChange={console.log}>{'Hello\nWorld!'}</Txt>
  //   ).toJSON()).toMatchSnapshot();
  // });
});
