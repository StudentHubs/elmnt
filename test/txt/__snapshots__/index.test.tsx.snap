// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`txt empty 1`] = `
<span
  className=" e1 e2 e3 e4 e5"
  style={
    Object {
      "display": "block",
    }
  }
>
  <span
    style={
      Object {
        "display": "block",
        "minHeight": 20,
        "padding": "1px 0px",
      }
    }
  >
    <span
      style={
        Object {
          "display": "block",
          "fontFamily": "inherit",
          "fontSize": 20,
          "lineHeight": "30px",
          "margin": "-6px 0px -6px",
        }
      }
    >
      
    </span>
  </span>
</span>
`;
